src/main/java/io/vertx/mqtt/MqttClientOptions.java

```diff
@@ -202,10 +206,25 @@ public String getWillTopic() {
   /**
    * @return will message content
    */
-  public String getWillMessage() {
+  @GenIgnore
+  public Buffer getWillMessage() {
     return willMessage;
   }
 
+  /**
+   * @return will message content as string
+   */
+  public String getWillMessageAsString() {
+    return getWillMessageAsString(StandardCharsets.UTF_8);
+  }
+
+  /**
+   * @return will message content as string
+   */
```

Comment:
I would add the `@param` in the javadoc to make the difference from the other overload.

Reply:
I have added this